src/pages/single-post-page/SinglePostPage.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from "react";
+import { NavBar, RightSection, PostCard } from "../../components";
+import { NavLink, useParams } from "react-router-dom";
+import { usePosts } from "../../features/posts/postSlice";
+import "./singlepostpage.css";
+import { addComment } from "../../features/posts/postSlice";
+import { useDispatch } from "react-redux";
+
+export function SinglePostPage() {
+  const [isLoading, setIsLoading] = useState(false);
+  const { posts } = usePosts();
+  const dispatch = useDispatch();
+  const [commentText, setCommentText] = useState("");
+  const { postId } = useParams();
+  const post = posts.find((post) => post._id === postId);
+
+  const addCommentToPost = (postId) => {
+    dispatch(addComment({ postId, comment: commentText }));
+    setCommentText((prevComment) => ({ ...prevComment, comment: "" }));
+  };
+
+  return (
+    <div className="homepage">
+      <NavBar />
+      <div className="homepage-section">
+        <div className="left-section">
```

Comment:
only if, left-section is common than you can also move it and create a component out of it

Reply:
It's not common. I have removed add post from it here.